py3-tensorflow-metadata.yaml

```diff
@@ -34,37 +41,69 @@ pipeline:
       repository: https://github.com/tensorflow/metadata
       tag: v${{package.version}}
 
-  - name: Python Build
-    runs: python setup.py build
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      runtime:
+        - py${{range.key}}-absl-py
+        - py${{range.key}}-googleapis-common-protos
+        - py${{range.key}}-protobuf
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+    test:
+      pipeline:
+        - name: Import Test
+          uses: python/import
+          with:
+            python: python${{range.key}}
+            import: ${{vars.import}}
+        - name: Basic Functionality Test
+          runs: |
+            echo "Testing ${{vars.pypi-package}} basic functionality..."
+            python${{range.key}} <<EOF-
+            from ${{vars.import}}.proto.v0 import schema_pb2
+            schema = schema_pb2.Schema()
+            schema.feature.add(name='example_feature')
+            assert schema.feature[0].name == 'example_feature', 'Basic feature addition failed'
+            print('${{vars.pypi-package}} basic functionality success')
+            EOF-
+        - uses: strip
 
-  - name: Python Install
-    runs: python setup.py install --prefix=/usr --root="${{targets.destdir}}"
-
-  - uses: strip
+  - name: py3-supported-${{vars.pypi-package}}
+    description: meta package providing ${{vars.pypi-package}} for supported python versions.
+    dependencies:
+      runtime:
+        - py3.10-${{vars.pypi-package}}
+        - py3.11-${{vars.pypi-package}}
+        - py3.12-${{vars.pypi-package}}
+        - py3.13-${{vars.pypi-package}}
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python3.10
+            import: ${{vars.import}}
+        - uses: python/import
+          with:
+            python: python3.11
+            import: ${{vars.import}}
+        - uses: python/import
+          with:
+            python: python3.12
+            import: ${{vars.import}}
+        - uses: python/import
+          with:
+            python: python3.13
+            import: ${{vars.import}}
 
 update:
   enabled: true
   manual: false
   github:
     identifier: tensorflow/metadata
     strip-prefix: v
-
```

Comment:
I'd prefer we keep a top level tests so that when the package with no version is installed we know *a* compatible version was installed.

Reply:
good call. adding that back actually highlighted that I'd missed a provides.